src/main/webapp/app/shared/subject/subject-detail.component.html

```diff
@@ -39,11 +42,14 @@ <h2><span [translate]="'managementPortalApp.subject.detail.title'"></span>: {{su
                 <span class="form-control-static">{{entry.key}}: {{entry.value}}</span>
             </div>
         </dd>
-
-        <dt><span [translate]="'managementPortalApp.subject.personName'"></span></dt>
-        <dd>{{subject.personName}}</dd>
-        <dt><span [translate]="'managementPortalApp.subject.dateOfBirth'"></span></dt>
-        <dd>{{subject.dateOfBirth | date:'yyyy-MM-dd' }}</dd>
+        <div *ngIf="!(siteSettings$ | async).hiddenSubjectFields.includes('person_name')">
+            <dt><span [translate]="'managementPortalApp.subject.personName'"></span></dt>
+            <dd>{{subject.personName}}"</dd>
+        </div>
+        <div *ngIf="!(siteSettings$ | async).hiddenSubjectFields.includes('date_of_birth')">
+            <dt><span [translate]="'managementPortalApp.subject.dateOfBirth'"></span></dt>
+            <dd>{{subject.dateOfBirth | date:'yyyy-MM-dd' }}</dd>
+        </div>
```

Comment:
To avoid a lot of subscriptions from a component, you can create a single subscription in a parent div and store and reuse the result:
```html
<div *ngIf="(siteSettings$ | async).hiddenSubjectFields as hiddenFields">
   ...
   <div *ngIf="!hiddenFields.includes('person_name')"...>...</div>
   <div *ngIf="!hiddenFields.includes('date_of_birth')"...>...</div
</div>
```
strangely this assignment only works in an ngIf, so the value must evaluate to true. Fortunately, objects and arrays always evaluate to true.